src/python/bcc/table.py

```diff
@@ -227,8 +227,8 @@ def _get_event_class(event_map):
                    'unsigned __int128' : (ct.c_ulonglong * 2),
                    'void *'            : ct.c_void_p }
 
-    # handle array types e.g. "int [16] foo"
-    array_type = re.compile(r"(.+) \[([0-9]+)\]$")
+    # handle array types e.g. "int [16]" or "char[16]"
+    array_type = re.compile(r"([^ ]+) ?\[([0-9]+)\]$")
```

Comment:
Does this match type like `struct foo[16]`?

Reply:
No, neither the original implementation.

Reply:
> No, neither the original implementation.

Okay, I see. I though `(.+) \[([0-9]+)\]$` may cover `struct foo [16]` with hoping `(.+) to cover `struct foo` but apparently this is not the case.

This functionality is used for python and we don't really have complicated python tools and that is probably reason we didn't hit `struct foo[16]` issue. I will merge this patch as it will fix a regression due to latest llvm. If we ever hit `struct foo[16]` issue later, we can address it then.